drivers/rf-transceiver/hmc630x/hmc630x.h

```diff
@@ -0,0 +1,304 @@
+#define HMC630X_EN_REFBUF		HMC630X_PARAM(17, NO_OS_BIT(3))
+#define HMC630X_EN_STICK_DIV		HMC630X_PARAM(17, NO_OS_BIT(2))
+#define HMC630X_EN_FBDIV_CML2CMOS	HMC630X_PARAM(17, NO_OS_BIT(1))
+#define HMC630X_EN_FBDIV		HMC630X_PARAM(17, NO_OS_BIT(0))
+
+/* ROW18 */
+#define HMC630X_EN_NB250M		HMC630X_PARAM(18, NO_OS_BIT(6))
+#define HMC630X_BYP_VCO_LDO		HMC630X_PARAM(18, NO_OS_BIT(5))
+#define HMC630X_EN_EXTLO		HMC630X_PARAM(18, NO_OS_BIT(4))
+#define HMC630X_EN_VCOPK		HMC630X_PARAM(18, NO_OS_BIT(3))
+#define HMC630X_EN_VCO			HMC630X_PARAM(18, NO_OS_BIT(2))
+#define HMC630X_EN_VCO_REG		HMC630X_PARAM(18, NO_OS_BIT(1))
+#define HMC630X_ENBAR_VCOGB		HMC630X_PARAM(18, NO_OS_BIT(0))
+
+/* ROW19 */
+#define HMC630X_REFSEL_SYNTHBG		HMC630X_PARAM(19, NO_OS_BIT(1))
+#define HMC630X_MUXREF			HMC630X_PARAM(19, NO_OS_BIT(0))
+
+/* ROW20 */
+#define HMC630X_FBDIV_CODE		HMC630X_PARAM(20, NO_OS_GENMASK(6, 0))
+
+/* ROW21 */
+#define HMC630X_REFSEL_VCOBG		HMC630X_PARAM(21, NO_OS_BIT(4))
+#define HMC630X_VCO_BIASTRIM		HMC630X_PARAM(21, NO_OS_GENMASK(3, 0))
+
+/* ROW22 */
+#define HMC630X_VCO_BANDSEL		HMC630X_PARAM(22, NO_OS_GENMASK(6, 0))
+
+/* ROW23 */
+#define HMC630X_ICP_BIASTRIM		HMC630X_PARAM(23, NO_OS_GENMASK(7, 5))
+#define HMC630X_VCO_OFFSET		HMC630X_PARAM(23, NO_OS_GENMASK(4, 0))
+
+/* ROW24 */
+#define HMC630X_LOCKDET			HMC630X_PARAM(24, NO_OS_BIT(3))
+#define HMC630X_DN			HMC630X_PARAM(24, NO_OS_BIT(2))
+#define HMC630X_UP			HMC630X_PARAM(24, NO_OS_BIT(1))
+#define HMC630X_CENTER			HMC630X_PARAM(24, NO_OS_BIT(0))
+
+/* ROW25 */
+#define HMC630X_VTUNE_FLASHP		HMC630X_PARAM(25, NO_OS_GENMASK(7, 0))
+
+/* ROW26 */
+#define HMC630X_VTUNE_FLASHN		HMC630X_PARAM(26, NO_OS_GENMASK(7, 0))
+
+/* ROW27 */
+#define HMC630X_TEMPS			HMC630X_PARAM(27, NO_OS_GENMASK(4, 0))
+
+/* Devices supported by this driver. */
+enum hmc630x_type {
+	HMC6300,
+	HMC6301,
+};
+
+/* Possible values for receiver LNA gain. */
+enum hmc6301_lna_gain {
+	HMC6301_LNA_GAIN_0dB,
+	HMC6301_LNA_GAIN_6dB,
+	HMC6301_LNA_GAIN_12dB,
+	HMC6301_LNA_GAIN_18dB
+};
+
+/* Possible baseband attenuation values. */
+enum hmc6301_bb_attn {
+	HMC6301_BB_ATTN_0dB,
+	HMC6301_BB_ATTN_6dB,
+	HMC6301_BB_ATTN_12dB,
+	HMC6301_BB_ATTN_18dB
+};
+
+/* Possible values for the external reference clock. */
+enum hmc630x_ref_clk {
+	HMC6300_REF_CLK_71p42857MHz,
+	HMC6300_REF_CLK_75MHz,
+	HMC6300_REF_CLK_142p8571MHz,
+	HMC6300_REF_CLK_154p2857MHz,
+};
+
+/* Initialization parameters for hmc6300_init(). */
+struct hmc630x_init_param {
+	enum hmc630x_type type;
+	enum hmc630x_ref_clk ref_clk; /* Reference clock value. */
+	struct no_os_gpio_init_param en; /* EN GPIO signal of the digital interface. */
+	struct no_os_gpio_init_param
+		clk; /* CLK GPIO signal of the digital interface. */
+	struct no_os_gpio_init_param
+		data; /* DATA GPIO signal of the digital interface. */
+	struct no_os_gpio_init_param
+		scanout; /* SCANOUT GPIO signal of the digital interface. */
+};
+
+struct hmc630x_dev;
+struct hmc630x_vco;
```

Comment:
Why are these defined in the source file instead of here?

Reply:
I guess it makes more sense to hide these from the application so that apps perform any read/write through API. This avoids situations where API users mess with device.parameter directly.